.github/workflows/schedule.py

```diff
@@ -75,7 +89,105 @@ def _load_issue_template(path: str) -> tuple[dict[str, str], str]:
         return front_matter, f.read()
 
 
+def main(templates: list[tuple[str, datetime]], dry_run: bool) -> None:
+    for template, create_date in templates:
+        cmd_lookup, cmd_create = gh_commands(template, create_date)
+        if dry_run:
+            log.info('Would be running a mock command that creates an issue')
+            print(' '.join(cmd_create))
+        else:
+            create_issue(cmd_create, cmd_lookup)
+
+
+def _create(template: str,
+            start: datetime,
+            period_days: int,
+            at_hour: int,
+            ) -> tuple[str, datetime] | None:
+    """
+    Return a tuple of the template and the datetime.now() instance that matches the
+    configured shcedule for the provided template. This logs the schedule check for
+    each, and it only returns what falls within the schedule window.
+    """
+    log.info('Checking configured schedule time window for %r', template)
+    if 0 == (now - start).days % period_days and now.hour == at_hour:
+        log.info('Scheduling %r, it falls within schedule window', template)
+        return template, now
+    else:
+        log.info('Current time is outside of the configured schedule window')
+
+
 if __name__ == '__main__':
     logging.basicConfig(level=logging.INFO,
                         format='%(asctime)s %(levelname)+7s %(name)s: %(message)s')
-    create_upgrade_issue()
+
+    args = sys.argv[1:]
```

Comment:
This is way too much code outside of a function. Most of our scripts have a very small  `if __name__ == '__main__'` body, for good reason.